Tests/UnitTests/Makefile

```diff
@@ -0,0 +1,145 @@
+# COLORS
+RED=\033[0;31m
+GREEN=\033[0;32m
+ORANGE=\033[0;33m
+BLUE=\033[0;34m
+PURPLE=\033[0;35m
+CYAN=\033[0;36m
+LIGHTGRAY=\033[0;37m
+DARKGRAY=\033[1;30m
+YELLOW=\033[0;33m
+NC=\033[0m # No Color
+
+# Build path
+BUILD_DIR = Objects
+
+######################################
+# source
+
+# C sources																			
+# Need to include test, unity, real source file for the test, mock c functions minus source file for test
+C_SOURCES = \
+Tests/Test_$(TEST).c \
+Unity/unity.c \
+$(filter-out $(wildcard ../../Tests/*/Src/$(TEST).c), $(wildcard ../../*/Src/$(TEST).c))	\
+$(wildcard Mocks/RTOS/os.c) \
+$(filter-out $(wildcard Mocks/*/Src/$(TEST).c), $(wildcard Mocks/*/Src/*.c))
+
+
+
+$(foreach src,$(C_SOURCES),$(info --- $(src)))
+
+#######################################
+# binaries
+#######################################
+
+CC = gcc
+AS = gcc -x assembler-with-cpp
+CP = objcopy
+SZ = size
+
+HEX = $(CP) -O ihex
+BIN = $(CP) -O binary -S
+
+SF = st-flash
+ 
+#######################################
+# C_INCLUDES
+#######################################
+
+#-iquote: used to stop the compiler from searching the current directory first. Another effect: -Idirs before it are only searched for "" files
+# Whatever I put after iquote doesn't seem to be found, but putting it in twice works. This will break #include-next, though
```

Comment:
Oh, these are notes on alternate solutions to the deprecated -I- flag. I should probably remove them.

-I- stops the compiler from searching the current directory first, which we want it to do so that it will find our mock headers before the real ones. -I- deprecated and asks not to be used, but we haven't found a good alternative. iquote is the most similar to an alternative but doesn't stop the compiler from searching the current directory, which is the feature we need. The notes are just results from messing around with iquote, though I don't really know what was actually going on.